### PR TITLE
[Catalog] Completed audit of swift forced unwrapping in catalog

### LINKED ISSUE #3984: Audit Swift force unwrapping in our catalog
As an MDC-iOS client, I want the catalog to include Swift examples that make use of the most current Swift best-practices.

---

Definition of done:

- Usage of force-unwrapping is replaced with recommended Swift patterns, such as guard statements.

---
#### Internal data
- Associated internal bug: [b/117179059](http://b/117179059)

### PR DESCRIPTION
**Pull request by GitPushPullLegs**
This work closes #3984.

This change audits the rest of the catalog swift examples force unwrapping. I had begun this process with PR #6117.

### CODE REVIEW

File: components/MaskedTransition/examples/MaskedTransitionTypicalUse.swift [hunk 1]
```diff
@@ -117,27 +119,29 @@ open class MaskedTransitionTypicalUseSwiftExample: UIViewController {
...
-    }
-
-    vc.view.autoresizingMask = target.autoresizingMask
-
-    // Customize the transition
-    let transitionController = MDCMaskedTransitionController(sourceView: fab)
-    if target.calculateFrame != nil {
-      transitionController.calculateFrameOfPresentedView = target.calculateFrame
-      vc.modalPresentationStyle = .custom
+    if let indexPathForSelectedRow = tableView.indexPathForSelectedRow {
```
Comment (yarneo) [inline components/MaskedTransition/examples/MaskedTransitionTypicalUse.swift]: tip: by changing this into a `guard let` and returning early you can remove the added block and indentation caused by the conditional.

**Review by yarneo — APPROVED** [review]
Thank you @joeyDevelops ! Left a minor tip for improvement.

File: components/MaskedTransition/examples/MaskedTransitionTypicalUse.swift (hunk 1 above)
Comment (GitPushPullLegs) [reply components/MaskedTransition/examples/MaskedTransitionTypicalUse.swift]: Oh yes, I like that idea.

File: components/MaskedTransition/examples/MaskedTransitionTypicalUse.swift [hunk 2]
```diff
@@ -117,26 +119,27 @@ open class MaskedTransitionTypicalUseSwiftExample: UIViewController {
 
   var transitionController: MDCMaskedTransitionController? = nil
   func didTapFab(fab: UIView) {
-    let target = targets[tableView.indexPathForSelectedRow!.row]
+    guard let indexPathForSelectedRow = tableView.indexPathForSelectedRow else { return }
+    let target = targets[indexPathForSelectedRow.row]
     let vc = target.viewControllerType.init()
-
+
```
Comment (yarneo) [inline components/MaskedTransition/examples/MaskedTransitionTypicalUse.swift]: thanks for committing the improvement! Just a few small things, there are now tabs here in the empty lines that should be reverted.

File: components/MaskedTransition/examples/MaskedTransitionTypicalUse.swift [hunk 3]
```diff
@@ -117,26 +119,27 @@ open class MaskedTransitionTypicalUseSwiftExample: UIViewController {
...
+    
     vc.view.autoresizingMask = target.autoresizingMask
-
+    
     // Customize the transition
     let transitionController = MDCMaskedTransitionController(sourceView: fab)
     if target.calculateFrame != nil {
-      transitionController.calculateFrameOfPresentedView = target.calculateFrame
-      vc.modalPresentationStyle = .custom
+        transitionController.calculateFrameOfPresentedView = target.calculateFrame
```
Comment (yarneo) [inline components/MaskedTransition/examples/MaskedTransitionTypicalUse.swift]: This should still be 2 spaces indentation and not 4 spaces

**Review by jverkoey — APPROVED** [review]
LGTM. In a follow-up PR we may want to explore the use of the `?? UIImage()` pattern introduced in this PR because, while it does resolve runtime crashes due to missing resources, I'm worried that it will cover up the fact that the resources might be missing.

Generally I'd like to encourage a failure to load an expected resource to be considered something that should be `assert`'d or otherwise made apparent so that it's clear that an expected resource is missing.

**Comment by jverkoey** [comment]
@joeyDevelops looks like you may need to rebase/merge your PR with the latest changes in develop.

**Pull request merged**